### PR TITLE
RMRK V2

### LINKED ISSUE #57: Export Minimark as subpackages
# What is the current solution

currently when someone wants to use `minimark` she or he will need to export functions like

```js
import { createCollection } from '@kodadot1/minimark'
```

It is good but we will have V2 #43 soon and I would cause chaos

# What I am trying to achieve

that we will have Subpackages

For RMRK v1

```js
import { createCollection } from '@kodadot1/minimark/v1'
```

For RMRK v2

```js
import { createCollection } from '@kodadot1/minimark/2'
```

For Common

```js
import { asSystemRemark } from '@kodadot1/minimark/common'
```

# I already hacked something

### but it is not ideal and I am missing CommonJS

[main...unmark](https://github.com/kodadot/packages/compare/main...unmark)

### PR DESCRIPTION
**Pull request by vikiival**
- feat: types for rmrk2.0 interaction
- feat: testing for LOCK and ACCEPT
- feat: add types for rmrk2.0.0 interactions
- feat: revert rmrk1.0 test file
- feat: types for json field
- feat: unwrap for v2 new implementation
- feat: testing suit for rmrk2.0
- feat: update field parser
- feat: reset skip
- fix: mis deleted line
- refactor: rename types
- refactor: move v2 code
- feat: BUY
- ref: interaction constants and type
- feat: V2 EMOTE
- feat: send list changeissuer
- fix: remove skip
- feat: CREATE
- feat: MINT
- feat: BURN
- feat: create InteractionV2
- feat: colleciton and nft types
- fix: add ACCEPT and missing field
- refactor: using same field name with rmrk standard
- refactor: using same field name with rmrk standard
- feat: create Interaction
- feat: test for create interaction
- feat: create collection
- feat: create nft
- feat: type for BASE entity
- feat: create BASE
- fix: expected test should be false
- refactor: check props for createCollection
- feat: sync test cases
- feat: error for create interaction
- fix: rename types and solidator
- fix: clean types
- fix: export all in one
- :truck: :test_tube: reorder tests
- :truck: moved v2 mocks into correct folder
- :bug: unreachable part of code
- :memo: do not forget
- :bug: use same types
- :label: Abstract NFT type
- :zap: V2 impl of some functions
- :test_tube: createCollection use new params
- :bug: transferable is binary only in v1
- :label: use type for created NFT
- :zap: use parameters for createNFT v2
- :see_no_evil: suite is good for small scripts
- :label: Base has proper type
- :zap: lower trim for base
- :truck: basic part
- :truck: rename parts
- :zap: toLowerCase
- :zap: makeBaseSymbol
- :truck: CreatedBASE to CreatedBase
- :zap: somethign
- :label: Ipfs Uri type
- :label: a bit of magick with Base
- :soap: optimized imports for create
- :bug: teplate literal are unknown to this version of typescript
- :bug: :see_no_evil: some residuals from merging
- :test_tube: create/v2
- :soap: :test_tube: polished tests
- :soap: linted RMRK v2
- :coffin: no more jest
- :soap: linted RMRK v1
- :wrench: lint-fix
- :soap: :test_tube: polished tests
- :bug: Symbol cannnot use - or .
- :wrench: tests without lint
- :bug: base symbol should be lowercased and underscored
- :wrench: test without coverage
- :test_tube: create passes
- :truck: named exports only
- :construction: :truck: moving code into packages
- :construction: :truck: shuffling types
- :construction: :truck: shuffling types v2
- :label: interactions are under file enumes
- :zap: beatutiful unwrap
- :label: Beatutiful types
- :construction: :truck: shuffling types
- :tada: shared created called make
- :zap: using make as abstraction for create
- :zap: make Interaction can has generic meta parametert
- :zap: create in V1 logically creates v1 interaction
- :zap: code will throw on max < 0
- :zap: code will throw on max < 0
- :zap: check if string is emppty
- :truck: moved identification to shared
- :test_tube: :truck: shuffling
- :package: updating exports
- :wrench: add compatible subpath imports, Close #57
- :wrench: pack
- :heavy_plus_sign: @rmrk-team tools
- :construction: :test_tube: testiing
- :bug: treat base symbol as UPPERCASED
- :zap: base symbol should be always lowercased YOLO
- :test_tube: unconsolidated cases in symbol
- :test_tube: magick

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Chore

### What's new?

- [ ] PR closes #<issue_number>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR
- [ ] I've written some unit tests 🧪

### Optional

- [ ] I found edge cases

### Had issue bounty label?

- [ ] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=<My_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address>)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### CODE REVIEW

**Comment by vikiival** [comment]
> it's happening 

Hidrasec-582x449

**Review by vikiival — COMMENTED** [review]
Uff

**Comment by vikiival** [comment]
If @roiLeo or @Jarsen136 can check code? 
I may forgot something

File: minimark/src/rmrk/v2/create.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    default:
+      throw new Error(`Unsupported action: ${action}`)
+  }
+}
+
+// DEV: not sure if trasferable should be
+export const createNFT = (index: number, collectionId: string, name: string | undefined, metadata: string, transferable = 1, royalty?: RoyaltyInfo): CreatedNFT => {
+  // checkProps(props)
+  // const { symbol, index, transferable = 1, collectionId, metadata } = props
+  const sn = toSerialNumber(index)
```
Comment (Matehoo) [inline minimark/src/rmrk/v2/create.ts:65]: In some places, we use `sn` and in others `serialNumber`. We should unify it across the whole app.

File: minimark/src/rmrk/shared/make.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    id: toCollectionId(caller, theSymbol),
+    symbol: theSymbol,
+    issuer: caller,
+    name: trim(name),
+    max,
+    metadata
+  }
+}
+
+export const makeInteraction = <I, T = string>(action: I, version: RemarkVersion = '1.0.0', objectId: string, meta: T): RemarkableString => {
```
Comment (Matehoo) [inline minimark/src/rmrk/shared/make.ts:21]: Why is the default `RemarkVersion = '1.0.0'`?

File: minimark/src/rmrk/v2/create.ts (hunk 1 above)
Comment (vikiival) [reply minimark/src/rmrk/v2/create.ts:65]: fixed some parts here d5a9a12

File: minimark/test/v2/remarkable/resource.test.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { describe, expect, it } from 'vitest'
+
+describe('RMRK2 Resource', () => {
+  it('should create add resource', () => {
+    expect(true).toBe(true)
+  })
+})
```
[the remark this answers is not in the document]
Comment (vikiival) [reply minimark/test/v2/remarkable/resource.test.ts:7]: Made follow up issue https://github.com/kodadot/packages/issues/119
